service/controller/azuremachine/handler/azuremachinemetadata/create.go

```diff
@@ -0,0 +1,41 @@
+package azuremachinemetadata
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/giantswarm/azure-operator/v5/pkg/helpers"
+	"github.com/giantswarm/azure-operator/v5/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, cr interface{}) error {
+	var err error
+	azureMachine, err := key.ToAzureMachine(cr)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// We need this for existing clusters, and we do it both in Cluster
+	// controller (clusterupgrade) and in AzureMachine controller, because we
+	// do not know which controller/handler will be executed first, and we need
+	// annotation set correctly in both places.
+	err = helpers.InitAzureMachineAnnotations(ctx, r.ctrlClient, r.logger, &azureMachine)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.ctrlClient.Update(ctx, &azureMachine)
+	if errors.IsConflict(err) {
+		r.logger.Debugf(ctx, "conflict trying to save object in k8s API concurrently", "stack", microerror.JSON(microerror.Mask(err)))
```

Comment:
Yo! That thingie doesn't work as expected. :grimacing: 

I'd propose to start discussion on adding `Warningf()` here: https://github.com/giantswarm/micrologger/blob/master/spec.go#L7 to get the desired outcome. :slightly_smiling_face: